gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java

```diff
@@ -480,7 +481,7 @@ private void showProjectLoadingDialog() {
     }
   }
 
-  public void showFeatureTypeDialog(Layer layer, Point point) {
+  private void showFeatureTypeDialog(Layer layer, Point point) {
     ArrayAdapter<String> arrayAdapter =
         new ArrayAdapter(getContext(), R.layout.project_selector_list_item, R.id.project_name);
     arrayAdapter.add(getString(R.string.point));
```

Comment:
Please define a constant which maps `FeatureType` to the string representation and use that here.
Also, it would be nicer to put all of this dialog related code into a separate class (like `AddFeatureDialogFragment`) and only handle the final selected option here.

Reply:
Yes but shall dialogFragment share a ViewModel with the HomeScreenFragment or shall it listen the click on dialog using interfaces as we only need the position it was clicked.

Reply:
A click listener should be sufficient.

Reply:
implemented.